src/main/java/se/diabol/jenkins/pipeline/DeliveryPipelineView.java

```diff
@@ -504,10 +506,38 @@ private Component getComponent(String name, AbstractProject firstJob, AbstractPr
 
     @Override
     public Collection<TopLevelItem> getItems() {
-        return (Collection)getOwnerItemGroup().getItems();
-    }
-
-    @Override
+        Set<TopLevelItem> jobs = Sets.newHashSet();
+	    addJobsFromComponentSpecs(jobs);
+	    addRegexpFirstJobs(jobs);
+	    return jobs;
+    }
+
+	private void addRegexpFirstJobs(Set<TopLevelItem> jobs) {
+		if (regexpFirstJobs == null) {
+			return;
+		}
+		for (RegExpSpec spec : regexpFirstJobs) {
+	        Map<String, AbstractProject> regexpJobs = getProjects(spec.getRegexp());
+	            for (Map.Entry<String, AbstractProject> entry : regexpJobs.entrySet()) {
+	               jobs.add((TopLevelItem) entry.getValue());
+	            }
+	    }
+	}
+
+	private void addJobsFromComponentSpecs(Set<TopLevelItem> jobs) {
+		if (componentSpecs == null) {
+			return;
+		}
+		for (ComponentSpec spec : componentSpecs) {
+		    AbstractProject first = getProject(spec.getFirstJob(), getOwnerItemGroup());
+		    AbstractProject last = getProject(spec.getLastJob(), getOwnerItemGroup());
+		    for (AbstractProject project : getAllDownstreamProjects(first, last, spec.getExcludeJobsRegex()).values()) {
```

Comment:
Maybe `getAllDownstreamProjects(first, last, spec.getExcludeJobsRegex()).values()` could be extracted to a variable?